django/utils/timesince.py

```diff
@@ -40,10 +40,15 @@ def timesince(d, now=None, reversed=False):
     if not now:
         now = datetime.datetime.now(utc if is_aware(d) else None)
 
-    delta = (d - now) if reversed else (now - d)
+    if reversed:
+        d, now = now, d
+    delta = now - d
 
     # Deal with leapyears by subtracing the number of leapdays
-    delta -= datetime.timedelta(calendar.leapdays(d.year, now.year))
+    leapdays = calendar.leapdays(d.year, now.year)
+    if leapdays != 0 and calendar.isleap(d.year) and d.date() > datetime.date(d.year, 2, 29):
```

Comment:
I don't see any test failures if `and d.date() > datetime.date(d.year, 2, 29)` is removed. Could you test test that boundary if it's needed?